src/core/worker.js

```diff
@@ -197,6 +197,138 @@ IPDFStreamRangeReader.prototype = {
 }
 
 /** @implements {IPDFStream} */
+function PDFNetworkStream(data, msgHandler) {
+  this.data = data;
+  this.msgHandler = msgHandler;
+  let source = data.source;
+  this._contentLength = source.length;
+  this._fullRequestReader = null;
+  this._rangeRequestReaders = [];
+}
+
+PDFNetworkStream.prototype = {
+  getFullReader() {
+    assert(!this._fullRequestReader);
+    this._fullRequestReader = new PDFNetworkStreamReader(this);
+    return this._fullRequestReader;
+  },
+
+  getRangeReader(begin, end) {
+    let reader = new PDFNetworkStreamRangeReader(begin, end, this);
+    this._rangeRequestReaders.push(reader);
+    return reader;
+  },
+
+  cancelAllRequests(reason) {
+    if (this._fullRequestReader) {
+      this._fullRequestReader.cancel(reason);
+    }
+    let readers = this._rangeRequestReaders.slice(0);
+    readers.forEach(function (reader) {
+      reader.cancel(reason);
+    });
+  },
+};
+
+/** @implements {IPDFStreamReader} */
+function PDFNetworkStreamReader(networkStream) {
+  this._msgHandler = networkStream.msgHandler;
+  this._data = networkStream.data;
+  this.onProgress = null;
+
+  this._contentLength = null;
+  this._isRangeSupported = false;
+  this._isStreamingSupported = false;
+  let queueingStrategy = { highWaterMark: 100,
```

Comment:
let's remove queueingStrategy from here and range reader, or make explicit highWaterMark:1 and size:()=>1